### PR TITLE
Suggest geocodes and send them with the new form request

### PR DESCRIPTION
**Pull request by attwad**
Screen Recording 2020-06-19 at 1 39 53 PM mov

Had to change the schema for lat lng to be numbers to accept "2" as a lat for example otherwise an error would be thrown.

Also only send non empty even fields because the mongo schema validator doesn't like null dates.

### CODE REVIEW

File: verification/curator-service/ui/cypress/integration/components/LinelistTableTest.spec.ts [hunk 1]
```diff
@@ -5,6 +5,10 @@ describe('Linelist table', function () {
         cy.login();
     });
 
+    afterEach(() => {
```
Comment (allysonjp715) [inline verification/curator-service/ui/cypress/integration/components/LinelistTableTest.spec.ts:8]: Is this needed? I don't see anything being seeded in these tests.

File: verification/curator-service/ui/cypress/integration/components/NewCaseForm.spec.ts [hunk 2]
```diff
@@ -72,13 +79,22 @@ describe('New case form', function () {
     });
 
     it('Does not add row on submission error', function () {
-        // Avoid geolocation fail, the failure should happen at the data service level.
-        cy.seedLocation({ name: '' });
+        // Avoid geolocation fail, the "Request failed" check below happen at the data service level.
```
Comment (allysonjp715) [inline verification/curator-service/ui/cypress/integration/components/NewCaseForm.spec.ts:82]: s/happen/happens

File: verification/curator-service/ui/src/components/new-case-form-fields/LocationForm.tsx [hunk 3]
```diff
@@ -1,28 +1,125 @@
...
-            </Scroll.Element>
-        );
-    }
+interface Values {
+    location: Loc;
+}
+function LocationForm(): JSX.Element {
+    const { values } = useFormikContext<Values>();
+    return (
+        // TODO: suggest and pass location.
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/new-case-form-fields/LocationForm.tsx:18]: Can be removed now

File: verification/curator-service/ui/src/components/new-case-form-fields/LocationForm.tsx [hunk 4]
```diff
@@ -1,28 +1,125 @@
...
+    React.useEffect(() => {
+        let active = true;
+
+        if (inputValue.trim() === '') {
+            setOptions(value ? [value] : []);
+            return undefined;
+        }
+
+        fetch({ q: inputValue }, (results?: Loc[]) => {
+            console.log('fetch results:', results);
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/new-case-form-fields/LocationForm.tsx:68]: Remove

File: verification/curator-service/ui/cypress/integration/components/LinelistTableTest.spec.ts (hunk 1 above)
Comment (attwad) [reply verification/curator-service/ui/cypress/integration/components/LinelistTableTest.spec.ts:8]: addCase does seed a location.

File: verification/curator-service/ui/cypress/integration/components/NewCaseForm.spec.ts (hunk 2 above)
Comment (attwad) [reply verification/curator-service/ui/cypress/integration/components/NewCaseForm.spec.ts:82]: done

File: verification/curator-service/ui/src/components/new-case-form-fields/LocationForm.tsx (hunk 3 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/new-case-form-fields/LocationForm.tsx:18]: yay

File: verification/curator-service/ui/src/components/new-case-form-fields/LocationForm.tsx (hunk 4 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/new-case-form-fields/LocationForm.tsx:68]: done

**Pull request merged**